packages/calcite-components/src/components/date-picker/date-picker.tsx

```diff
@@ -532,12 +554,13 @@ export class DatePicker implements LocalizedComponent, LoadableComponent, T9nCom
 
   private setEndDate(date: Date): void {
     const startDate = this.getStartDate();
-    const newEndDate = date ? setEndOfDay(date) : date;
+    this.userChangeRangeValue = true;
     this.value = [dateToISO(startDate), dateToISO(date)];
     this.valueAsDate = [startDate, date];
-    this.mostRecentRangeValue = newEndDate;
-    this.calciteDatePickerRangeChange.emit();
-    this.activeEndDate = date || null;
+    // avoid emitting change event when user select stat date beyond end date
```

Comment:
Can we use a self-documenting var for this instead? Applies to similar comments.